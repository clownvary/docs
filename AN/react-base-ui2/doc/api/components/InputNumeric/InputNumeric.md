## InputNumeric  extends InputBase
---
InputNumeric component allows you to input numeric values only.

InputNumeric has 3 types of style: they are: 'number', 'currency' and 'percent'.

This can be determined by changing the type prop.

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
type|string|NumericType.NUMBER|false|The type of the InputNumeric.
value|string &#124; number &#124; bool|null|false|The current value.
min|number|-9999999999.99|false|Determines the minimal value that can be entered.
max|number|9999999999.99|false|Determines the maximum value that can be entered.
showGroup|bool|false|false|Indicates whether the thousands group separator will be inserted between  each digital group. (number of digits in thousands group depends on the selected Culture)
decimals|number|2|false|Indicates the custom setting for decimal places to display.
increment|number|1|false|Determines how much to increase/decrease the input field.
allowKeySpin|bool|true|false|Allow increment or decrement by typing the up or down keys.
textAlign|string|'right'|false|Determines the left or right alignment of text.
onValueChange|func|identity|false|A function called when the value of the input is changed.
allowBlank|bool|true|false|Determines whether blank text is displayed after selecting all and pressing delete.
listClass|string|'input-numeric-list'|false|&ensp;