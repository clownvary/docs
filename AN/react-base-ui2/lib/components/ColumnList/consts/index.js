'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ColumnType = require('./ColumnType');var _ColumnType2 = _interopRequireDefault(_ColumnType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);
var _SortOrder = require('../../../consts/SortOrder');var _SortOrder2 = _interopRequireDefault(_SortOrder);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  SortOrder: _SortOrder2.default,
  ColumnType: _ColumnType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];