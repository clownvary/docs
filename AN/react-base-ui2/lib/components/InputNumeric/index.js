'use strict';Object.defineProperty(exports, "__esModule", { value: true });exports.NUM_MAX = exports.NUM_MIN = exports.NumericType = exports.InputNumericField = exports.InputNumericProps = exports.InputNumeric = undefined;var _InputNumeric = require('./InputNumeric');var _InputNumeric2 = _interopRequireDefault(_InputNumeric);
var _NumericType = require('../../consts/NumericType');var NumericType = _interopRequireWildcard(_NumericType);
var _validation = require('../../services/validation');function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

var InputNumericField = (0, _validation.createField)(_InputNumeric2.default);exports.


InputNumeric = _InputNumeric2.default;exports.
InputNumericProps = _InputNumeric.InputNumericProps;exports.
InputNumericField = InputNumericField;exports.
NumericType = NumericType;exports.
NUM_MIN = _InputNumeric.NUM_MIN;exports.
NUM_MAX = _InputNumeric.NUM_MAX;exports.default =


_InputNumeric2.default;