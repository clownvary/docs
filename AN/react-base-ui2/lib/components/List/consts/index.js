'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _ListType = require('./ListType');var _ListType2 = _interopRequireDefault(_ListType);
var _SelectionMode = require('../../../consts/SelectionMode');var SelectionMode = _interopRequireWildcard(_SelectionMode);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  ListType: _ListType2.default,
  SelectionMode: SelectionMode };module.exports = exports['default'];