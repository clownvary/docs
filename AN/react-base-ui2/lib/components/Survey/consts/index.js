'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _QuestionType = require('./QuestionType');var QuestionType = _interopRequireWildcard(_QuestionType);
var _InputType = require('./InputType');var InputType = _interopRequireWildcard(_InputType);function _interopRequireWildcard(obj) {if (obj && obj.__esModule) {return obj;} else {var newObj = {};if (obj != null) {for (var key in obj) {if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key];}}newObj.default = obj;return newObj;}}exports.default =

{
  QuestionType: QuestionType,
  InputType: InputType };module.exports = exports['default'];