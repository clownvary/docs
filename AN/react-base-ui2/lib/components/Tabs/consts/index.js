'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _TabsClasses = require('./TabsClasses');var _TabsClasses2 = _interopRequireDefault(_TabsClasses);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =


{
  TabsClasses: _TabsClasses2.default };module.exports = exports['default'];