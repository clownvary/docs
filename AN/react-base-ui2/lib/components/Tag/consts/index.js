'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _TagSize = require('./TagSize');var _TagSize2 = _interopRequireDefault(_TagSize);
var _TagType = require('./TagType');var _TagType2 = _interopRequireDefault(_TagType);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  TagSize: _TagSize2.default,
  TagType: _TagType2.default };module.exports = exports['default'];