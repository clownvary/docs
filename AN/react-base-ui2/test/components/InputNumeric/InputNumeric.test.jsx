import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import InputNumeric, { NumericType } from '../../../src/components/InputNumeric';
import { KeyCode } from '../../../src/consts';

jest.useFakeTimers();

const setup = props => mount(<InputNumeric {...props} />);

describe('components/InputNumeric', () => {
  test('rendering', () => {
    let wrapper = setup({ value: '999' });
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper = setup({ value: 99, type: NumericType.PERCENT });
    expect(toJSON(wrapper)).toMatchSnapshot();

    wrapper = setup({ value: 1299, type: NumericType.CURRENCY });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('get and set value', () => {
    const instance = setup().instance();
    instance.value = 100;
    expect(instance.value).toBe(100);
  });

  test('props change', () => {
    let wrapper = setup({ min: 0, max: 10, value: 1 });
    let instance = wrapper.instance();

    instance.updateRange = jest.fn();
    wrapper.setProps({ min: 5 });
    expect(instance.updateRange).toBeCalledWith(5, 10);

    instance.updateCultureContext = jest.fn();
    wrapper.setProps({ decimals: 5 });
    expect(instance.updateCultureContext).toBeCalledWith('decimals', 5);
    wrapper.setProps({ showGroup: true });
    expect(instance.updateCultureContext).toBeCalledWith('showGroup', true);
    wrapper.setProps({ locale: 'test' });
    expect(instance.updateCultureContext).toBeCalledWith();

    instance.setValue = jest.fn();
    wrapper.setProps({ value: 9.9 });
    expect(instance.setValue).toBeCalledWith(9.9);

    wrapper = setup({ min: 1, max: 10, value: 5 });
    instance = wrapper.instance();

    instance.setValue = jest.fn();
    wrapper.setProps({ max: 4 });
    expect(instance.setValue).toBeCalledWith(5);
    wrapper.setProps({ min: 6, max: 10 });
    expect(instance.setValue).toBeCalledWith(5);
    wrapper.setProps({ decimals: 5 });
    expect(instance.setValue).toBeCalledWith(5, true);
    wrapper.setProps({ locale: 'test' });
    expect(instance.setValue).toBeCalledWith(5, true);

    wrapper = setup();
    wrapper.setProps({ value: 10 });
  });

  test('key press preview', () => {
    let wrapper = setup();
    let instance = wrapper.instance();
    let input = wrapper.find('input');
    input.simulate('keyPress', { nativeEvent: { char: '$' } });
    expect(instance.input.value).toBe('');
    input.simulate('keyPress', { nativeEvent: { char: '%' } });
    expect(instance.input.value).toBe('');
    input.simulate('keyPress', { nativeEvent: { char: 'a' } });
    expect(instance.input.value).toBe('');
    input.simulate('keyPress', { nativeEvent: { char: '.' } });
    expect(instance.input.value).toBe('');
    input.simulate('keyPress', { nativeEvent: { char: '9' } });
    expect(instance.input.value).toBe('9.00');
    input.simulate('keyPress', { nativeEvent: { char: '+' } });
    expect(instance.input.value).toBe('9.00');
    input.simulate('keyPress', { nativeEvent: { char: '-' } });
    expect(instance.input.value).toBe('-9.00');
    input.simulate('keyPress', { nativeEvent: { char: '(' } });
    expect(instance.input.value).toBe('9.00');
    input.simulate('keyPress', { nativeEvent: { char: ')' } });
    expect(instance.input.value).toBe('-9.00');
    input.simulate('keyPress', { nativeEvent: { char: '.' } });
    expect(instance.input.value).toBe('-9.00');
    instance.selection.start = 3;
    instance.selection.end = 3;
    input.simulate('keyPress', { nativeEvent: { char: '.' } });
    expect(instance.input.value).toBe('-9.00');

    wrapper = setup({ value: -10 });
    instance = wrapper.instance();
    input = wrapper.find('input');

    expect(instance.input.value).toBe('-10.00');
    input.simulate('keyPress', { nativeEvent: { char: '+' } });
    expect(instance.input.value).toBe('10.00');
    input.simulate('keyPress', { nativeEvent: { char: ')' } });
    expect(instance.input.value).toBe('-10.00');


    wrapper = setup({ value: 0 });
    instance = wrapper.instance();
    input = wrapper.find('input');

    input.simulate('keyPress', { nativeEvent: { char: '+' } });
    expect(instance.input.value).toBe('0.00');


    input.simulate('keyPress', { nativeEvent: { char: ')' } });
    expect(instance.input.value).toBe('0.00');
  });

  test('blur', () => {
    const wrapper = setup({ value: 0, min: 10, max: 20 });
    const instance = wrapper.instance();
    const input = wrapper.find('input');
    input.simulate('blur');
    expect(instance.value).toBe(10);

    wrapper.setProps({ value: 30 });
    input.simulate('blur');
    expect(instance.value).toBe(20);

    wrapper.setProps({ value: 15 });
    input.simulate('blur');
    expect(instance.value).toBe(15);
  });

  test('spins', () => {
    let wrapper = setup({ value: 10 });
    let instance = wrapper.instance();
    let input = wrapper.find('input');

    input.simulate('keyDown', { keyCode: KeyCode.UP });
    expect(instance.value).toBe(11);

    jest.runAllTimers();
    input.simulate('keyDown', { keyCode: KeyCode.DOWN });
    expect(instance.value).toBe(10);

    wrapper = setup();
    instance = wrapper.instance();
    input = wrapper.find('input');

    input.simulate('keyDown', { keyCode: KeyCode.DOWN });
    expect(instance.value).toBe(null);

    jest.runAllTimers();
    input.simulate('keyDown', { keyCode: KeyCode.UP });
    expect(instance.value).toBe(null);

    wrapper.setProps({ increment: 999999999999 });
    instance.doSpin();
    expect(instance.value).toBe(null);

    wrapper.setProps({ increment: 1, decimals: 0, value: 0 });
    instance.doSpin();
    expect(instance.value).toBe(-1);
  });

  test('deletion', () => {
    const wrapper = setup({ value: 99 });
    const instance = wrapper.instance();
    const input = wrapper.find('input');
    input.simulate('keyDown', { keyCode: KeyCode.UP });
    expect(instance.value).toBe(100);

    jest.runAllTimers();
    input.simulate('keyDown', { keyCode: KeyCode.BACKSPACE });
    expect(instance.value).toBe(100);

    jest.spyOn(instance, 'isSelectAll').mockImplementation(() => true);
    jest.runAllTimers();
    input.simulate('keyDown', { keyCode: KeyCode.BACKSPACE });
    expect(instance.value).toBe(null);

    jest.spyOn(instance, 'allowEdit').mockImplementation(() => false);
    wrapper.setProps({ value: 10, allowBlank: false });
    instance.deleteSelection();
    expect(instance.value).toBe(10);

    instance.allowEdit.mockRestore();
    instance.deleteSelection();
    expect(instance.value).toBe(0);

    instance.isSelectAll.mockRestore();
    wrapper.setProps({ value: 100 });
    jest.runAllTimers();
    input.simulate('keyDown', { keyCode: KeyCode.DELETE });
    expect(instance.value).toBe(0);
  });

  test('work with items', () => {
    const items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, true];
    let wrapper = setup({ items, value: 1, showTrigger: true });
    let instance = wrapper.instance();
    const trigger = wrapper.find('a.button-toggler');
    trigger.simulate('click');
    const listItems = document.querySelectorAll('.an-popup li');
    expect(listItems.length).toBe(11);
    const containerName = Object.keys(window).filter(name => /_popup_container_/.test(name))[0];
    ReactDOM.unmountComponentAtNode(window[containerName]);

    instance.onListSelected([]);
    expect(instance.value).toBe(1);

    instance.onListSelected([1]);
    expect(instance.value).toBe(2);

    instance.onListSelected([10]);
    expect(instance.value).toBe(1);

    instance.onListSelected([11]);
    expect(instance.value).toBe(1);

    instance.onListSelected([-1]);
    expect(instance.value).toBe(1);


    wrapper = setup({ items: [1, 2, 3], type: NumericType.PERCENT, value: 10, decimals: -1 });
    instance = wrapper.instance();
    expect(instance.getValue()).toBe(10);

    wrapper = setup({ items: null, type: NumericType.PERCENT, value: 10, showTrigger: true });
    instance = wrapper.instance();
    expect(instance.findInList()).toEqual([-1]);
  });

  test('misc', () => {
    const wrapper = setup({ value: 10, min: 5, max: 15 });
    const instance = wrapper.instance();

    jest.spyOn(instance, 'resetCursor');
    jest.spyOn(instance, 'select');
    jest.spyOn(instance, 'isBlank').mockImplementationOnce(() => true);
    instance.onIMEBreakThrough();
    expect(instance.resetCursor).toBeCalled();
    instance.onIMEBreakThrough();
    expect(instance.select).toBeCalled();

    instance.updateRange();
    expect(wrapper.state('stateClassName')).toBe('');

    instance.setValue(true);
    expect(instance.value).toBe(1);
    instance.setValue(false);
    expect(instance.value).toBe(0);
    jest.spyOn(instance.textProvider, 'setValue').mockImplementationOnce(() => false);
    instance.setValue(20, true);
    expect(instance.value).toBe(20);

    instance.textProvider = null;
    instance.setValue(30);
    expect(instance.value).toBe('20.00');
  });
});
