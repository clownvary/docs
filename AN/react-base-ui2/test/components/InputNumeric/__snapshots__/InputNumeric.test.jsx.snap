// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/InputNumeric rendering 1`] = `
<InputNumeric
  allowBlank={true}
  allowKeySpin={true}
  allowMouseSpin={false}
  ariaLabel="Input box"
  className=""
  currency="USD"
  data-qa-id=""
  decimals={2}
  disabled={false}
  icon=""
  iconHint=""
  id=""
  increment={1}
  items={Array []}
  listClass="input-numeric-list"
  listClassName=""
  listConfig={Object {}}
  listMaxHeight="300px"
  listPopupOptions={Object {}}
  listWidth="auto"
  max={9999999999.99}
  min={-9999999999.99}
  name=""
  onBlur={[Function]}
  onFocus={[Function]}
  onListClose={[Function]}
  onListOpen={[Function]}
  onTextChange={[Function]}
  onValueChange={[Function]}
  placeHolder=""
  prefix="an"
  showClear={false}
  showGroup={false}
  showSpinner={false}
  showTrigger={false}
  showTrigger2={false}
  size="md"
  textAlign="right"
  triggerIcon="icon-chevron-down"
  triggerIcon2="icon-chevron-down"
  triggerIconHint=""
  triggerIconHint2=""
  triggerIconToggle="icon-chevron-up"
  triggerIconToggle2="icon-chevron-up"
  value="999"
>
  <div
    className="an-input-numeric input-group buttons-0 input-group--md"
  >
    <input
      aria-label="Input box"
      className="input input-group__field textalign-right input__field"
      data-qa-id=""
      disabled={false}
      id=""
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      placeholder=""
      type="text"
    />
  </div>
</InputNumeric>
`;

exports[`components/InputNumeric rendering 2`] = `
<InputNumeric
  allowBlank={true}
  allowKeySpin={true}
  allowMouseSpin={false}
  ariaLabel="Input box"
  className=""
  currency="USD"
  data-qa-id=""
  decimals={2}
  disabled={false}
  icon=""
  iconHint=""
  id=""
  increment={1}
  items={Array []}
  listClass="input-numeric-list"
  listClassName=""
  listConfig={Object {}}
  listMaxHeight="300px"
  listPopupOptions={Object {}}
  listWidth="auto"
  max={9999999999.99}
  min={-9999999999.99}
  name=""
  onBlur={[Function]}
  onFocus={[Function]}
  onListClose={[Function]}
  onListOpen={[Function]}
  onTextChange={[Function]}
  onValueChange={[Function]}
  placeHolder=""
  prefix="an"
  showClear={false}
  showGroup={false}
  showSpinner={false}
  showTrigger={false}
  showTrigger2={false}
  size="md"
  textAlign="right"
  triggerIcon="icon-chevron-down"
  triggerIcon2="icon-chevron-down"
  triggerIconHint=""
  triggerIconHint2=""
  triggerIconToggle="icon-chevron-up"
  triggerIconToggle2="icon-chevron-up"
  type="percent"
  value={99}
>
  <div
    className="an-input-numeric input-group buttons-0 input-group--md"
  >
    <input
      aria-label="Input box"
      className="input input-group__field textalign-right input__field"
      data-qa-id=""
      disabled={false}
      id=""
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      placeholder=""
      type="text"
    />
  </div>
</InputNumeric>
`;

exports[`components/InputNumeric rendering 3`] = `
<InputNumeric
  allowBlank={true}
  allowKeySpin={true}
  allowMouseSpin={false}
  ariaLabel="Input box"
  className=""
  currency="USD"
  data-qa-id=""
  decimals={2}
  disabled={false}
  icon=""
  iconHint=""
  id=""
  increment={1}
  items={Array []}
  listClass="input-numeric-list"
  listClassName=""
  listConfig={Object {}}
  listMaxHeight="300px"
  listPopupOptions={Object {}}
  listWidth="auto"
  max={9999999999.99}
  min={-9999999999.99}
  name=""
  onBlur={[Function]}
  onFocus={[Function]}
  onListClose={[Function]}
  onListOpen={[Function]}
  onTextChange={[Function]}
  onValueChange={[Function]}
  placeHolder=""
  prefix="an"
  showClear={false}
  showGroup={false}
  showSpinner={false}
  showTrigger={false}
  showTrigger2={false}
  size="md"
  textAlign="right"
  triggerIcon="icon-chevron-down"
  triggerIcon2="icon-chevron-down"
  triggerIconHint=""
  triggerIconHint2=""
  triggerIconToggle="icon-chevron-up"
  triggerIconToggle2="icon-chevron-up"
  type="currency"
  value={1299}
>
  <div
    className="an-input-numeric input-group buttons-0 input-group--md"
  >
    <input
      aria-label="Input box"
      className="input input-group__field textalign-right input__field"
      data-qa-id=""
      disabled={false}
      id=""
      name=""
      onBlur={[Function]}
      onChange={[Function]}
      onDragEnter={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onMouseUp={[Function]}
      onPaste={[Function]}
      placeholder=""
      type="text"
    />
  </div>
</InputNumeric>
`;
